dev-qt/qtbase/files/qtbase-6.8.2-cross.patch

```diff
@@ -0,0 +1,37 @@
+When cross-compiling, CMake needs to find the build host's Qt6CoreTools. It
+therefore prepends QT_HOST_PATH, which is /usr, to CMAKE_FIND_ROOT_PATH. The
+problem is that CMAKE_FIND_ROOT_PATH is only a hint, not a definitive
+location. Just below, CMake's find_package is usually told to look in
+/usr/${CHOST}/usr/lib/cmake and /usr/lib/cmake when cross-compiling. Since
+both of these are under /usr, it chooses the former instead of the latter.
+It then ends up trying to execute non-native Qt binaries.
+
+We can avoid this problem by setting CMAKE_FIND_ROOT_PATH to a more precise
+location. All the Qt6 modules are installed under /usr/lib/cmake, represented
+by the __qt_find_package_host_qt_path variable, so we can point it there.
+
+find_package has two modes, module mode and config mode. No mode is
+explicitly chosen in this case, so it tries both. In module mode, it would
+use a module called FindQt6*.cmake, but no such module exists. It is
+therefore safe to assume config mode, which involves the files under
+/usr/lib/cmake.
+
+See the isSameDirectoryOrSubDirectory() call in CMake's
+cmFindCommon::RerootPaths() function for exactly where this goes wrong.
+
+Chewi
+
+https://bugs.gentoo.org/950314
+
+diff -Naur a/cmake/QtConfig.cmake.in b/cmake/QtConfig.cmake.in
+--- a/cmake/QtConfig.cmake.in	2024-11-14 11:02:40.000000000 +0000
++++ b/cmake/QtConfig.cmake.in	2025-03-02 14:29:42.190183608 +0000
```

Comment:
Please run `scrub-patch` from iwdevtools on it then feel free to merge. Thanks.

I *would* prefer if we didn't modify installed cmake files,  always hard to tell what kind of effect it may have some other things like e.g. qt-creator doing cross-compiling at runtime, but this particular change looks harmless enough.